SimPEG/regularization/vector.py

```diff
@@ -917,9 +917,12 @@ def f_m(self, m):
         numpy.ndarray
             The regularization kernel function evaluated for the model provided.
         """
-        a = self.amplitude(m)
+        fm = (
+            sp.block_diag([self.cell_gradient] * self.n_comp)
```

Comment:
Don’t do a block diag here (it is slow), reshape the `a` matrix into a (-1, 3) array and then multiply it by cell_gradient.